src/overlays/actors/ovl_En_Tanron3/z_en_tanron3.c

```diff
@@ -28,39 +36,380 @@ const ActorInit En_Tanron3_InitVars = {
+                          5.0f, 0);
+        }
+    }
+    if (this->acCollider.base.acFlags & AC_HIT) {
+        this->acCollider.base.acFlags &= ~AC_HIT;
+        if (this->deathTimer == 0) {
+            this->deathTimer = 15;
+            this->fogTimer = 15;
+            EnTanron3_SetupKill(this, globalCtx);
+            sGyorg->unk_324 = 20;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tanron3/func_80BB91D4.s")
+void EnTanron3_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnTanron3* this = THIS;
+    s16 i;
+    Vec3f splashPos;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tanron3/func_80BB9288.s")
+    if (KREG(63) == 0) {
+        this->timer++;
+        for (i = 0; i < TIMER_MAX; i++) {
+            if (this->workTimer[i] != 0) {
+                this->workTimer[i]--;
+            }
+        }
+        if (this->deathTimer != 0) {
+            this->deathTimer--;
+        }
+        if (this->fogTimer != 0) {
+            this->fogTimer--;
+        }
+        this->actionFunc(this, globalCtx);
+        Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 10.0f, 10.0f, 20.0f, 5);
+        if (((this->actor.prevPos.y < this->waterSurfaceYPos) && (this->waterSurfaceYPos <= this->actor.world.pos.y)) ||
+            ((this->actor.prevPos.y > this->waterSurfaceYPos) && (this->waterSurfaceYPos >= this->actor.world.pos.y))) {
+            splashPos.x = this->actor.world.pos.x;
+            splashPos.y = this->waterSurfaceYPos + 10.0f;
+            splashPos.z = this->actor.world.pos.z;
+            EffectSsGSplash_Spawn(globalCtx, &splashPos, NULL, NULL, 1, 500);
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_OUT_OF_WATER);
+        }
+    }
+    EnTanron3_CheckCollisions(this, globalCtx);
+    Collider_UpdateCylinder(&this->actor, &this->atCollider);
+    Collider_UpdateCylinder(&this->actor, &this->acCollider);
+    CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->atCollider.base);
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->acCollider.base);
+    if ((s8)sGyorg->actor.colChkInfo.health <= 0 && this->actionFunc != EnTanron3_Kill) {
+        EnTanron3_SetupKill(this, globalCtx);
+        this->workTimer[TIMER_PICK_DIRECTION_OR_DIE] = 0;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tanron3/func_80BB9308.s")
+s32 EnTanron3_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                               Actor* thisx) {
+    EnTanron3* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tanron3/EnTanron3_Update.s")
+    if (limbIndex == 1) {
+        rot->y += this->bodyRotation;
+    }
+    if (limbIndex == 3) {
+        rot->y += this->tailRotation;
+    }
+    if (limbIndex == 4) {
+        rot->y += this->trunkRotation;
+    }
+    return 0;
```

Comment:
```suggestion
    return false;
```
This is a boolean that determines if the limb should be drawn.